### ISSUE TITLE
HTTPS cURL request fails at 340th request when used in a recursive function

### ISSUE THREAD

**Issue by tschoonen** [php/doc-en#1757]

**Comment by tschoonen** [comment]
> > However, I wouldn't know how a programmer could be made aware of this without having intricate knowledge of this particular problem.
> 
> Well, the best we can do is to improve the documentation. Particularly, the `curl_close()` page should mention how to actually release a `CurlHandle` (not only that the function no longer has any effect). And it seems prudent to mention there, that reusing of handles is preferable over creating and destroying the handles repeatedly.

That does seem like the best approach. However, shouldn't the cURL extension maybe give more a more informative error message when this occurs? Such as 'too many open files', assuming that is the case. As it stands now it would be quite hard to deduce why cURL failed to perform the request.

**Comment by cmb69** [comment]
> However, shouldn't the cURL extension maybe give more a more informative error message when this occurs?

From what I can tell, we cannot influence that; the error is raised by libcurl. Trying to guess why libcurl failed doesn't look like a sensible approach. And after all, this is very much an edge-case; 400 requests are probably uncommon per se, but keeping the handles open is even more so.

**Comment by tschoonen** [comment]
Fair enough. Thanks for the helpful comments, I will relay this to my client.
